File: src/main/java/com/autosale/shop/service/UserService.java
package com.autosale.shop.service;

import com.autosale.shop.model.User;
import org.springframework.security.core.userdetails.UserDetailsService;

import java.util.List;

public interface UserService extends UserDetailsService {
    List<User> findAll();

    User findById(int id);

    Integer create(User user);

    void edit(User user);

    int delete(int id);

    User findByUsername(String username);

    User returnVerifiedUser(String username, String password);
}

File: src/main/java/com/autosale/shop/service/UserService.kt
package com.autosale.shop.service

import com.autosale.shop.model.User
import org.springframework.security.core.userdetails.UserDetailsService

interface UserService : UserDetailsService {
    fun findAll(): List<User>
    fun findById(id: Int): User
    fun create(user: User): Int
    fun edit(user: User)
    fun delete(id: Int): Int
    fun findByUsername(username: String): User
    fun getVerifiedUser(username: String, password: String): User
}
